packages/wizard/src/core/Wizard.ts

```
@@ -67,6 +68,28 @@ import type { GlobalInterceptorHandler } from '../types/type-wizard-global-flags
+ * ensuring that only the essential plugin registration, event listening, and command parsing interfaces are available.
+ *
+ * @template NameToContext - The mapping type from command names to their context, defining the supported commands and their context types.
+ * @template GlobalFlags - The type of global flags supported by the CLI, defaulting to the built-in flags.
+ *
+ * @example
+ * // Used for type constraints on the return value of plugin setup
+ * const cli: WizardWithUse = createWizard({ ... });
+ * cli.use(myPlugin).on('build', handler).parse();
+ */
+export type WizardWithUse<
+  NameToContext extends CommandNameToContext = {},
```

ESLint

[error] The `{}` ("empty object") type allows any non-nullish value, including literals like `0` and `""`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.

(@typescript-eslint/no-empty-object-type)